Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNode.java

```diff
@@ -179,32 +134,17 @@ private String getFunctionHttpTriggerUrl() {
         if (functionApp.getRuntime().getOperatingSystem() != OperatingSystem.WINDOWS) {
             return getAdminHttpTriggerUrl();
         }
-        final Map<String, String> keyMap = listFunctionKeys();
-        final String key = keyMap.values().stream().filter(StringUtils::isNotBlank)
-                .findFirst().orElse(getFunctionMasterKey());
+        final String key = functionApp.listFunctionKeys(this.name).values().stream().filter(StringUtils::isNotBlank)
+                .findFirst().orElse(functionApp.getMasterKey());
         return String.format(HTTP_TRIGGER_URL_WITH_CODE, functionApp.hostName(), this.name, key);
     }
 
     private String getAdminHttpTriggerUrl() {
-        return String.format(HTTP_TRIGGER_URL_WITH_CODE, functionApp.hostName(), this.name,
-                getFunctionMasterKey());
-    }
-
-    private HttpPost getFunctionTriggerRequest() {
-        final String targetUrl = String.format(NONE_HTTP_TRIGGER_URL, functionApp.hostName(), this.name);
-        final HttpPost request = new HttpPost(targetUrl);
-        request.setHeader("x-functions-key", getFunctionMasterKey());
-        request.setHeader("Content-Type", "application/json");
-        return request;
-    }
-
-    private String getFunctionMasterKey() {
-        // todo: imply with app service library
-        return StringUtils.EMPTY;
+        return String.format(HTTP_TRIGGER_URL_WITH_CODE, functionApp.hostName(), this.name, functionApp.getMasterKey());
     }
 
-    private Map<String, String> listFunctionKeys() {
-        // todo: imply with app service library
-        return Collections.EMPTY_MAP;
+    @RequiredArgsConstructor
+    static class TriggerRequest {
+        private final String input;
     }
```

Comment:
can we user only String to replace this class ?

Reply:
trigger api in sdk will serialize input object into json map, so we could not pass the value string directly which will miss the key